cmd/hubagent/workload/setup.go

```diff
@@ -105,12 +105,17 @@ func SetupControllers(ctx context.Context, wg *sync.WaitGroup, mgr ctrl.Manager,
 		}
 	}
 
-	disabledResourceConfig := utils.NewDisabledResourceConfig()
+	disabledResourceConfig := utils.NewResourceConfig(true)
 	if err := disabledResourceConfig.Parse(opts.SkippedPropagatingAPIs); err != nil {
 		// The program will never go here because the parameters have been checked
 		return err
 	}
 
+	allowedResourceConfig := utils.NewResourceConfig(false)
```

Comment:
only one of them can be set. You don't need to maintain two configurations. Please see the comments below.